pkg/apiserver/experiment/experiment.go

```diff
@@ -500,8 +759,53 @@ func (s *Service) listExperiments(c *gin.Context) {
 	c.JSON(http.StatusOK, data)
 }
 
-// TODO: need to be implemented
-func (s *Service) getExperimentDetail(c *gin.Context) {}
+// @Summary Get detailed information about the specified chaos experiment.
+// @Description Get detailed information about the specified chaos experiment.
+// @Tags experiments
+// @Produce json
+// @Param namespace path string true "namespace"
+// @Param name path string true "name"
+// @Param kind path string true "kind" Enums(PodChaos, IoChaos, NetworkChaos, TimeChaos, KernelChaos, StressChaos)
+// @Success 200 ExperimentInfo
+// @Router /api/experiments/detail/{kind}/{namespace}/{name} [GET]
+// @Failure 400 {object} utils.APIError
+// @Failure 500 {object} utils.APIError
+func (s *Service) getExperimentDetail(c *gin.Context) {
+	kind := c.Param("kind")
+	ns := c.Param("namespace")
+	name := c.Param("name")
+
+	var (
+		ok   bool
+		info ExperimentInfo
+		err  error
+	)
+	if _, ok = v1alpha1.Kinds[kind]; !ok {
+		c.Status(http.StatusBadRequest)
+		_ = c.Error(utils.ErrInvalidRequest.New(kind + " is not supported"))
+		return
+	}
+	switch kind {
+	case "PodChaos":
+		info, err = s.getPodChaosDetail(ns, name)
+	case "IoChaos":
+		info, err = s.getIoChaosDetail(ns, name)
+	case "NetworkChaos":
+		info, err = s.getNetworkChaosDetail(ns, name)
+	case "TimeChaos":
+		info, err = s.getTimeChaosDetail(ns, name)
+	case "KernelChaos":
+		info, err = s.getKernelChaosDetail(ns, name)
+	case "StressChaos":
+		info, err = s.getStressChaosDetail(ns, name)
+	}
```

Comment:
I prefer using `v1alpha1.KindXXX` here instead of plain string

Reply:
OK, i will edit it.